flixel/util/FlxPath.hx

```diff
@@ -207,12 +191,21 @@ class FlxPath implements IFlxDestroyable
 	
 	/**
 	 * Internal function for moving the object along the path.
-	 * Generally this function is called automatically by preUpdate().
 	 * The first half of the function decides if the object can advance to the next node in the path,
 	 * while the second half handles actually picking a velocity toward the next node.
 	 */
 	public function update(elapsed:Float):Void
 	{
+		if (object == null)
+			return;
+		
+		if (_firstUpdate)
```

Comment:
Both approaches are hacky, it's a bit late to avoid tight coupling between these two objects. Your implementation introduces a bug where the user's immovable value could get clobbered by the Path if they're not aware of the inner implementation.

The way to avoid the tight coupling would be to take a completely different approach to this problem. Avoiding tighter coupling is not a good excuse for leaving this bug in.